crates/bevy_animation/src/lib.rs

```diff
@@ -478,12 +545,25 @@ fn apply_animation(
     parents: &Query<(Option<With<AnimationPlayer>>, Option<&Parent>)>,
     children: &Query<&Children>,
 ) {
-    if let Some(animation_clip) = animations.get(&animation.animation_clip) {
-        if !paused {
+    let Some(animation_clip_handle) = &animation.animation_clip else {
+        return;
+    };
+
+    if let Some(animation_clip) = animations.get(animation_clip_handle) {
+        // Only update the elapsed time while the player is not paused and the animation is not complete.
+        // We don't return early because set_elapsed() may have been called on the animation player.
+        if !animation.finished() && !paused {
             animation.elapsed += time.delta_seconds() * animation.speed;
         }
         let mut elapsed = animation.elapsed;
-        if animation.repeat {
+
+        if (elapsed > animation_clip.duration && animation.speed > 0.0)
+            || (elapsed < 0.0 && animation.speed < 0.0)
+        {
+            animation.completions += 1;
+        }
```

Comment:
this doesn't work to count completions, it will never go to more than 1 which means the `RepeatAnimation::Count` mode won't work

Reply:
Very good point. This has been updated to use `seek_time`, rather than `elapsed` - I've expanded on this in (updated) PR comment :+1: